lib/stack_master/commands/apply.rb

```diff
@@ -62,12 +63,33 @@ def create_or_update_stack
       end
 
       def create_stack
+        upload_files
+        if use_change_set?
+          create_stack_by_change_set
+        else
+          create_stack_directly
+        end
+      end
+
+      def use_change_set?
+        @options.on_failure.nil?
+      end
+
+      def create_stack_by_change_set
+        @change_set = ChangeSet.create(stack_options.merge(change_set_type: 'CREATE'))
+        halt!(@change_set.status_reason) if @change_set.failed?
+        @change_set.display(StackMaster.stdout)
         unless ask?('Create stack (y/n)? ')
-          failed!("Stack creation aborted")
+          cf.delete_stack(stack_name: stack_name)
+          halt!('Stack creation aborted')
         end
-        upload_files
-        on_failure = @config.stack_defaults['on_failure'] || @options.on_failure
-        cf.create_stack(stack_options.merge({tags: proposed_stack.aws_tags, on_failure: on_failure}))
+        execute_change_set
+      end
+
+      def create_stack_directly
+        failed!('Stack creation aborted') unless ask?('Create stack (y/n)? ')
+        on_failure = @options.on_failure || 'ROLLBACK'
```

Comment:
This doesn't need a default because `create_stack_directly` is only called when `on_failure` is set